synapse/media/_base.py

```diff
@@ -674,6 +676,10 @@ def _parseparam(s: bytes) -> Generator[bytes, None, None]:
         s = s[end:]
 
 
+class ConsumerStopProducingError(Exception):
+    """A consumer asked us to stop producing"""
```

Comment:
The exception name itself is missing the slight context that the consumer *asked* for this.

Maybe:

```suggestion
class ConsumerRequestedStopError(Exception):
    """A consumer asked us to stop producing"""
```

Also fine as-is if we can't come up with something better.